executors/cpp/coll.cpp

```diff
@@ -70,57 +75,41 @@ const string test_collator(json_object *json_in)  //
   u_unescape(opt_source, source, 100);
   u_unescape(opt_target, target, 100);
 
-
-  if (locale_obj) {
-    has_locale = true;
-  } else {
-    has_locale = false;
-  }
-
-  bool ignore_punctuation = false;
   json_object *ignore_obj = json_object_object_get(json_in, "ignorePunctuation");
 
   if (ignore_obj) {
-    ignore_punctuation = true;
-    status = U_ZERO_ERROR;
     ucol_setMaxVariable(coll, UCOL_REORDER_CODE_PUNCTUATION, &status);
   }
 
+  const int32_t unspecified_length = -1;
   bool coll_result = true;
-  UCollationResult result = ucol_strcoll(coll,
-                                         source, -1,
-                                         target, -1);
+  UCollationResult result = ucol_strcoll(
+      coll,
+      source, unspecified_length,
+      target, unspecified_length);
 
-  // The json result.
+  // The json test output.
   json_object *return_json = json_object_new_object();
   json_object_object_add(return_json, "label", label_obj);
 
+  int64_t numeric_result = int64_t(result);
   if (result != UCOL_LESS) {
     coll_result = false;
-    // Include date for the failing test
-    json_object_object_add(return_json,
-                           "s1",
-                           json_object_new_string(string1.c_str()));
-    json_object_object_add(return_json,
-                           "s2",
-                           json_object_new_string(string2.c_str()));
-    if (result == UCOL_EQUAL) {
-      json_object_object_add(return_json,
-                             "compare",
-                             json_object_new_int64(0));
-    } else {
-      json_object_object_add(return_json,
-                             "compare",
-                             json_object_new_int64(1));
-    }
+
+    // Include data compared in the failing test
+    json_object_object_add(
+        return_json, "s1", json_object_new_string(string1.c_str()));
+    json_object_object_add(
+        return_json, "s2", json_object_new_string(string2.c_str()));
+
+    // What was the actual returned value?
+    json_object_object_add(
+        return_json, "compare", json_object_new_int64(numeric_result));
```

Comment:
yay, looking good (by using `numeric_result` as the arg to `json_object_new_int64`)